app/dashboard/src/layouts/Drive.tsx

```diff
@@ -154,14 +154,21 @@ export default function Drive(props: DriveProps) {
   }, [dispatchAssetListEvent])
 
   const doCreateProject = React.useCallback(
-    (templateId: string | null = null, templateName: string | null = null) => {
+    (
+      templateId: string | null = null,
+      templateName: string | null = null,
+      onCreated?: (project: backendModule.CreatedProject) => void,
+      onError?: () => void
+    ) => {
```

Comment:
callback mutation?

Reply:
just call the callback directly

Reply:
it would be possible to refactor all "create asset" events into a single event containing an `AnyAsset` and a callback that updates its internal state when it's created - but i'm not sure whether it's an ideal solution

Reply:
ok so upon further thinking i have some ideas on how to refactor things, but i feel like they don't really belong in this PR. if you have any less invasive solutions that'd work for this callback specifically, please do make suggestions